R/processes.R

```diff
@@ -0,0 +1,156 @@
+#' @title Define model processes
+#' @description
+#' create_processes, defines the functions which describe how each individual's
+#' states and variables change over time.
+#'
+#' It lists processes from `infection.R`, `mosquito_emergence.R` and
+#' `mortality.R`; and then exposes them to the model through
+#' `bind_process_to_model`
+#' @param individuals, a list of individuals in the model
+#' @param states, a list of states in the model
+#' @param variables, a list of variables and constants in the model
+#' @param parameters, a list of model parameters
+create_processes <- function(individuals, states, variables, parameters) {
+
+  list(
+    create_ageing_process(individuals$human, variables$age),
+
+    # ======
+    # States
+    # ======
+
+    # Untreated Progression
+    create_fixed_probability_state_change_process(
+      individuals$human,
+      states$I,
+      states$D,
+      1 - parameters$ft
+    ),
+    # Treatment
+    create_fixed_probability_state_change_process(
+      individuals$human,
+      states$I,
+      states$Treated,
+      parameters$ft
+    ),
+    # Asymptomatic Progression
+    create_fixed_probability_state_change_process(
+      individuals$human,
+      states$D,
+      states$A,
+      parameters$rd
+    ),
+    # Subpatient Progression
+    create_fixed_probability_state_change_process(
```

Comment:
Is there super-infection in this model, i.e. is it possible to go from "U" back to "A", or are flows currently only in one direction?